### PR TITLE
Fix small things

### LINKED ISSUE #338: 💄 📝 🚸 Small improvements
Some functions have very spartan documentation. List them here for improvement (🚑 = important):

- [ ] `add_reactions` 

Some functions need to be changed a little bit:

- [x] `Reaction` let reaction equation dicts also do `Dict{String, Int}` as opposed to `Dict{String, Float}` only
- [x] `change_bounds!` also doesn't accept Ints and gives a terrible error message
- [x] `add_objective!` - returns the last assigned value, which looks weird, make it return nothing
- [x] `flux_summary` - prints unbounded reactions
- [x] `load_model(StandardModel,...)` the conversion is slow, make it faster
- [x] `save_json_model` is painfully slow - the file shows up but is empty for 10s of seconds before things actually get written to it... it makes you think something has gone wrong. 🚑 
- [x] `is_mass_balanced` should issue a warning if a metabolite does not have a formula and you are checking if a reaction is mass balanced - false negatives could be generated...
- [ ] `remove_reaction(!)` is in place for `StandardModel` and returns a new model for `CoreModel` this makes the function signature slightly irritating to use since for the one model type you need a bang and for the other you don't.. maybe add in place versions for the `CoreModel` functions? 

Note: I will add more as I use COBREXA - don't address this in a PR yet!

### PR DESCRIPTION
**Pull request by stelmo**
Closes #338

### CODE REVIEW

File: src/base/types/Reaction.jl [hunk 1]
```diff
@@ -1,3 +1,4 @@
+using Base: Float64
```
Comment (exaexa) [inline src/base/types/Reaction.jl:1]: Why this?

File: src/base/utils/Reaction.jl [hunk 2]
```diff
@@ -61,7 +61,10 @@ function is_mass_balanced(model::StandardModel, rxn::Reaction)
     atom_balances = Dict{String,Float64}() # float here because stoichiometry is not Int
     for (met, stoich) in rxn.metabolites
         atoms = metabolite_formula(model, met)
-        isnothing(atoms) && continue # ignore blanks
+        if isnothing(atoms) # ignore blanks 
+            @warn "$met has no atoms associated with it."
```
Comment (exaexa) [inline src/base/utils/Reaction.jl:65]: there are the switchable logging functions to report this

File: src/base/types/Reaction.jl [hunk 3]
```diff
@@ -50,24 +51,95 @@ mutable struct Reaction
...
-        Reaction(id; metabolites = metabolites, lb = lb, ub = ub)
+"""
+Reaction(
+    id::String,
+    metabolites::Dict{String,Union{Int, Float64}},
+    dir = :bidirectional;
+    default_bound = _constants.default_reaction_bound,
+)
+
+Convenience constructor for `Reaction`.
```
Comment (exaexa) [inline src/base/types/Reaction.jl:64]: maybe add the fact that this one is created from a dictionary of metabolites and the second one from a vector

**Review by exaexa — CHANGES_REQUESTED** [review]

File: src/base/types/Reaction.jl (hunk 1 above)
Comment (stelmo) [reply src/base/types/Reaction.jl:1]: I took this out, I didn't add it, no idea how it got into master to begin with

Comment (exaexa) [reply src/base/types/Reaction.jl:1]: but this PR is adding it?

File: src/base/utils/Reaction.jl (hunk 2 above)
Comment (stelmo) [reply src/base/utils/Reaction.jl]: is this correct?

Comment (stelmo) [reply src/base/utils/Reaction.jl]: like the updated code

File: src/base/types/Reaction.jl (hunk 1 above)
Comment (stelmo) [reply src/base/types/Reaction.jl:1]: Nope, it should not be added. I think VScode is automatically adding things like this.. It is very irritating :/ I deleted it

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 5bf2e159 is formatted properly.

**Comment by stelmo** [comment]
This also closes #370

File: src/base/logging/log.jl [hunk 4]
```diff
@@ -56,3 +56,5 @@ end
 @_make_logging_tag models "model-related messages"
 @_make_logging_tag io "messages and warnings from model input/output"
 @_make_logging_tag perf "performance-related tracing information"
+@_make_logging_tag utilities "Utility function related logging information"
+log_utilities(true) # switch log on
```
[the remark this answers is not in the document]
Comment (stelmo) [reply src/base/logging/log.jl:60]: removed utilities logger and made the atoms thing part of models

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 66d87d4c is formatted properly.

File: src/base/logging/log.jl (hunk 4 above)
[the remark this answers is not in the document]
Comment (stelmo) [reply src/base/logging/log.jl:60]: Hmm although I think they should be on by default - most people won't look at the logs unless they are reminded that they exist

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 363acb5b is formatted properly.

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit f98dda12 requires formatting!

Required formatting changes summary:
``` src/base/utils/Reaction.jl | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)
```

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit aa3c3385 requires formatting!

Required formatting changes summary:
``` src/base/types/SBMLModel.jl | 42 +++++++++++++++++++++---------------------
 1 file changed, 21 insertions(+), 21 deletions(-)
```

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit f98dda12 requires formatting!

Required formatting changes summary:
``` src/base/utils/Reaction.jl | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)
```

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 82ef81ad is formatted properly.

**Review by exaexa — APPROVED** [review]

**Pull request merged**